### ISSUE TITLE
Facing issue while running API after migrating to v27

### ISSUE THREAD

**Issue by manci22lc** [ehrbase/ehrbase#1137]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Environment information

EHRbase version:  0.27.4
openEHR_SDK version:  1.29.0
Archie version: 3.3.0
PostgreSQL version: PostgreSQL 15.1
Java Runtime version: Eclipse Adoptium 17.0.7+7
OS: Linux amd64 5.10.102.1-microsoft-standard-WSL2

### Steps to reproduce

1. We have current version 23 running in our environment which has data.
2. Migrate-> change ehrbase image to latest version 0.27.4
3. Start the application 
4. Call ehrId creation API : {{ehrcore-baseurl}}/ehrbase/rest/openehr/v1/ehr

### Expected behavior

EhrId should be created: 
{
    "system_id": {
        "_type": "HIER_OBJECT_ID",
        "value": "4ee8c182-bb5b-4e2c-9857-3c865b329bb3"
    },
    "ehr_id": {
        "_type": "HIER_OBJECT_ID",
        "value": "7d37eba5-39c1-4edc-a11f-065247976359"
    },
    "ehr_status": {
        "_type": "EHR_STATUS",
        "name": {
            "_type": "DV_TEXT",
            "value": "EHR Status"
        },
        "subject": {
            "_type": "PARTY_SELF"
        },
        "archetype_node_id": "openEHR-EHR-EHR_STATUS.generic.v1",
        "uid": {
            "_type": "HIER_OBJECT_ID",
            "value": "dec2ee4d-2def-4c90-bde0-8d157ac1c38f::local.ehrbase.org::1"
        },
        "is_modifiable": true,
        "is_queryable": true
    },
    "time_created": {
        "_type": "DV_DATE_TIME",
        "value": "2023-06-13T10:25:59.150Z"
    }
}

### Actual result

Getting error:

{
    "error": "Internal Server Error",
    "message": "An internal error has occurred. Please contact your administrator."
}

### Further information

Logs :
2023-06-13 15:38:37 2023-06-13 10:08:37.481  INFO [] 8 --- [nio-8080-exec-2] o.s.web.servlet.DispatcherServlet        : Completed initialization in 5 ms
2023-06-13 15:39:21 2023-06-13 10:09:21.183 ERROR [traceId=373c4601dbb04fda] 8 --- [nio-8080-exec-3] o.ehrbase.rest.DefaultExceptionHandler   : 
2023-06-13 15:39:21 
2023-06-13 15:39:21 org.ehrbase.api.exception.InternalServerException: Could not create an EHR with given parameters.
2023-06-13 15:39:21 at org.ehrbase.service.EhrServiceImp.create(EhrServiceImp.java:160)
2023-06-13 15:39:21 at org.ehrbase.service.EhrServiceImp$$FastClassBySpringCGLIB$$36900d55.invoke(<generated>)
2023-06-13 15:39:21 at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
2023-06-13 15:39:21 at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)
2023-06-13 15:39:21 at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
2023-06-13 15:39:21 at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
2023-06-13 15:39:21 at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:102)
2023-06-13 15:39:21 at org.ehrbase.plugin.AbstractPluginAspect.proceed(AbstractPluginAspect.java:70)
2023-06-13 15:39:21 at org.ehrbase.plugin.AbstractPluginAspect.proceedWithPluginExtensionPoints(AbstractPluginAspect.java:139)
2023-06-13 15:39:21 at org.ehrbase.plugin.AbstractPluginAspect.proceedWithPluginExtensionPoints(AbstractPluginAspect.java:110)
2023-06-13 15:39:21 at org.ehrbase.plugin.EhrPluginAspect.aroundCreateEhr(EhrPluginAspect.java:68)
2023-06-13 15:39:21 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-06-13 15:39:21 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
2023-06-13 15:39:21 at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
2023-06-13 15:39:21 at java.base/java.lang.reflect.Method.invoke(Unknown Source)
2023-06-13 15:39:21 at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634)
2023-06-13 15:39:21 at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624)
2023-06-13 15:39:21 at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72)
2023-06-13 15:39:21 at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2023-06-13 15:39:21 at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
2023-06-13 15:39:21 at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
2023-06-13 15:39:21 at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
2023-06-13 15:39:21 at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
2023-06-13 15:39:21 at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2023-06-13 15:39:21 at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
2023-06-13 15:39:21 at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
2023-06-13 15:39:21 at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2023-06-13 15:39:21 at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
2023-06-13 15:39:21 at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
2023-06-13 15:39:21 at org.ehrbase.service.EhrServiceImp$$EnhancerBySpringCGLIB$$54786fd5.create(<generated>)
2023-06-13 15:39:21 at org.ehrbase.rest.openehr.OpenehrEhrController.createEhr(OpenehrEhrController.java:98)
2023-06-13 15:39:21 at org.ehrbase.rest.openehr.OpenehrEhrController$$FastClassBySpringCGLIB$$2675ab1.invoke(<generated>)
2023-06-13 15:39:21 at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
2023-06-13 15:39:21 at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)
2023-06-13 15:39:21 at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
2023-06-13 15:39:21 at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
2023-06-13 15:39:21 at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:89)
2023-06-13 15:39:21 at org.ehrbase.tenant.DefaultTenantAspect.action(DefaultTenantAspect.java:88)
2023-06-13 15:39:21 at org.ehrbase.tenant.DefaultTenantAspect.action(DefaultTenantAspect.java:71)
2023-06-13 15:39:21 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-06-13 15:39:21 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
2023-06-13 15:39:21 at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
2023-06-13 15:39:21 at java.base/java.lang.reflect.Method.invoke(Unknown Source)
2023-06-13 15:39:21 at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634)
2023-06-13 15:39:21 at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624)
2023-06-13 15:39:21 at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72)
2023-06-13 15:39:21 at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:175)
2023-06-13 15:39:21 at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
2023-06-13 15:39:21 at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
2023-06-13 15:39:21 at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
2023-06-13 15:39:21 at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
2023-06-13 15:39:21 at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
2023-06-13 15:39:21 at org.ehrbase.rest.openehr.OpenehrEhrController$$EnhancerBySpringCGLIB$$b946dc8f.createEhr(<generated>)
2023-06-13 15:39:21 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-06-13 15:39:21 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
2023-06-13 15:39:21 at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
2023-06-13 15:39:21 at java.base/java.lang.reflect.Method.invoke(Unknown Source)
2023-06-13 15:39:21 at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
2023-06-13 15:39:21 at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)
2023-06-13 15:39:21 at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)
2023-06-13 15:39:21 at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895)
2023-06-13 15:39:21 at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
2023-06-13 15:39:21 at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
2023-06-13 15:39:21 at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1072)
2023-06-13 15:39:21 at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:965)
2023-06-13 15:39:21 at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
2023-06-13 15:39:21 at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
2023-06-13 15:39:21 at javax.servlet.http.HttpServlet.service(HttpServlet.java:555)
2023-06-13 15:39:21 at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
2023-06-13 15:39:21 at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
2023-06-13 15:39:21 at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
2023-06-13 15:39:21 at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
2023-06-13 15:39:21 at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
2023-06-13 15:39:21 at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
2023-06-13 15:39:21 at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
2023-06-13 15:39:21 at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:96)
2023-06-13 15:39:21 at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
2023-06-13 15:39:21 at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
2023-06-13 15:39:21 at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
2023-06-13 15:39:21 at org.ehrbase.application.web.LoggingContextFilter.doFilterInternal(LoggingContextFilter.java:55)
2023-06-13 15:39:21 at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
2023-06-13 15:39:21 at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:102)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
2023-06-13 15:39:21 at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
2023-06-13 15:39:21 at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
2023-06-13 15:39:21 at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
2023-06-13 15:39:21 at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481)
2023-06-13 15:39:21 at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
2023-06-13 15:39:21 at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
2023-06-13 15:39:21 at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
2023-06-13 15:39:21 at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
2023-06-13 15:39:21 at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:390)
2023-06-13 15:39:21 at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
2023-06-13 15:39:21 at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:926)
2023-06-13 15:39:21 at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1791)
2023-06-13 15:39:21 at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
2023-06-13 15:39:21 at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
2023-06-13 15:39:21 at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
2023-06-13 15:39:21 at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
2023-06-13 15:39:21 at java.base/java.lang.Thread.run(Unknown Source)
2023-06-13 15:39:21 Caused by: java.lang.IllegalArgumentException: Cannot translate a null SQLException
2023-06-13 15:39:21 at org.springframework.util.Assert.notNull(Assert.java:201)
2023-06-13 15:39:21 at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:68)
2023-06-13 15:39:21 at org.ehrbase.service.PersistenceConfig$ExceptionTranslator.exception(PersistenceConfig.java:51)
2023-06-13 15:39:21 at org.jooq.impl.ExecuteListeners.exception(ExecuteListeners.java:304)
2023-06-13 15:39:21 at org.jooq.impl.Tools.exception(Tools.java:2802)
2023-06-13 15:39:21 at org.jooq.impl.Tools.fetchOne(Tools.java:2738)
2023-06-13 15:39:21 at org.jooq.impl.Tools.fetchOne(Tools.java:2706)
2023-06-13 15:39:21 at org.jooq.impl.DefaultDSLContext.fetchOne(DefaultDSLContext.java:4886)
2023-06-13 15:39:21 at org.jooq.impl.DefaultDSLContext.fetchOptional(DefaultDSLContext.java:5155)
2023-06-13 15:39:21 at org.ehrbase.dao.access.interfaces.I_ConceptAccess.loadConcept(I_ConceptAccess.java:136)
2023-06-13 15:39:21 at org.ehrbase.dao.access.interfaces.I_ConceptAccess.lambda$getConceptByConceptId$0(I_ConceptAccess.java:110)
2023-06-13 15:39:21 at org.ehrbase.service.KnowledgeCacheService.getConceptByConceptId(KnowledgeCacheService.java:577)
2023-06-13 15:39:21 at org.ehrbase.dao.access.interfaces.I_ConceptAccess.getConceptByConceptId(I_ConceptAccess.java:109)
2023-06-13 15:39:21 at org.ehrbase.dao.access.interfaces.I_ConceptAccess.fetchContributionChangeType(I_ConceptAccess.java:102)
2023-06-13 15:39:21 at org.ehrbase.dao.access.jooq.ContributionAccess.setAuditDetailsValues(ContributionAccess.java:395)
2023-06-13 15:39:21 at org.ehrbase.dao.access.jooq.EhrAccess.commit(EhrAccess.java:549)
2023-06-13 15:39:21 at org.ehrbase.service.EhrServiceImp.create(EhrServiceImp.java:158)
2023-06-13 15:39:21 ... 113 common frames omitted
2023-06-13 15:39:21

**Comment by manci22lc** [comment]
@vidi42 

1. Running ehrbase as  docker container and its my local system, and have used same ehrbase version published on hub, so not building anything ourselves for testing.
2. Post EHR has empty request body, that is how we have been using in older versions.

**Comment by vidi42** [comment]
@manci22lc thanks for the feedback. It seems to work even with an empty EHR body. I've tried it on multiple systems and could not reproduce it with 0.27.4.

Our suspicion is that it may have something to do with the `concept` table and some data inside the table.
First it would help to find out what SQL query may cause this and then follow from there, so I would ask you to do some debugging and try and see which query is causing the exception.

You can print JOOQ queries by setting this property in your `application.yml` file `logging.level.org.jooq.tools.LoggerListener=DEBUG`

**Comment by vidi42** [comment]
@manci22lc I noticed a similar exception on one of my local environments after upgrading it.
The issue was caused by the system missing the ehrbase_restricted user.
Since you said you migrated from 0.23.0 I expect this can be a cause for you as well, because the non-privileged user was introduced in 0.24.0

Have a look at the [UPDATING.md](multi-tenant-plugin-1.5.1-SNAPSHOT.jar) guide and make sure to adapt your Docker image and also to execute the `add_restricted_user` script.

I hope it helps.
